tcms/static/js/testrun_actions.js

```diff
@@ -311,39 +298,35 @@ var updateCaseRunStatus = function(e) {
     } else {
       fireEvent(link, 'click');
     }
-  };
 
-  // Add comment
-  if (parameters['comment'] != '') {
-    // Reset the content to loading
-    var ajax_loading = getAjaxLoading();
-    ajax_loading.id = 'id_loading_' + parameters['case_id'];
-    container.html(ajax_loading);
-    var c = jQ('<div>');
-
-    if (parameters['value'] != '') {
-        jsonRPC('TestExecution.add_comment', [object_pk, parameters['comment']], function(data){
-            updateCommentsCount(parameters['case_id'], true);
-        });
-    } else {
-        jsonRPC('TestExecution.add_comment', [object_pk, parameters['comment']], function(data){
-            updateCommentsCount(parameters['case_id'], true);
-            callback();
-        });
+    const submitButtons = updateForm.find('.submit_button')
 
+    bindSubmitButtons(updateForm, data.status_id, submitButtons, executionId, caseId)
+  });
+};
+
+function bindSubmitButtons(updateForm, currentStatusId, submitButtons, executionId, caseId) {
+  submitButtons.each((_index, b) => {
+    const submitButton = $(b)
+    const executionStatusId = parseInt(submitButton.attr('statusId'))
+
+    submitButton.unbind('click')
+
+    if (executionStatusId === currentStatusId) {
+      submitButton.css('opacity', '0.25')
+    } else {
+      submitButton.css('opacity', '1')
+      submitButton.bind('click', () => updateExecutionStatus(updateForm, executionId, caseId, executionStatusId))
     }
-  }
 
-  // Update the object when changing the status
-  if (parameters['value'] != '') {
-    // Reset the content to loading
-    var ajax_loading = getAjaxLoading();
-    ajax_loading.id = 'id_loading_' + parameters['case_id'];
-    container.html(ajax_loading);
-    updateRunStatus([object_pk], value, callback);
-  }
-};
+  })
+}
 
+function addComment(commentForm, executionId, caseId) {
```

Comment:
For `commentForm` you are passing the textarea element, not the surrounding form itself so you may as well name it properly.